File: frontend/src/api/audit.ts
import { api } from './client'
import { AuditLog } from '../types'

export async function fetchAuditLog(limit = 150): Promise<AuditLog[]> {
  const { data } = await api.get<AuditLog[]>('/audit', { params: { limit } })
  return data
}

File: backend/app/api/audit.py
from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.db.session import get_db
from app.models.audit_log import AuditLog
from app.models.user import User
from app.schemas.audit_log import AuditLogResponse

router = APIRouter(prefix='/audit', tags=['audit'])


@router.get('', response_model=list[AuditLogResponse])
def list_audit(
    limit: int = 150,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    _ = current_user
    return list(
        db.scalars(
            select(AuditLog)
            .order_by(AuditLog.created_at.desc())
            .limit(min(limit, 500))
        ).all()
    )
